### PR TITLE
Zombies upgrade

### PR DESCRIPTION
**Pull request by FluffyGhoster**
Added 3 subtypes of zombie.
Subtype of zombie: bull, very tanky, slow moving, short sprint and low stamina.
Subtype of zombie: hunter, very fast and sensitive to sounds, can see through walls other mobs, can leap far and often, natural climber, relatively squishy, cannot infect, attacks have a chance to electric shock, good against IPCs.
Subtype of zombie: thino, relatively slow, can leap after a small delay and land sending people in a radius of 1 to fly except one, that gets red grabbed, punches relatively hard.
Zombie producing chemical grenade added.
Turning into zombie is now ~40% faster than before.
Zombies now have a limited stamina.
Normal zombies don't see through walls anymore, only the hunter do.
Some dmdoc and spacemandmm constrains added.

Balance is up to experimentation, it should be workable, but needs field test to tweak it as expected.

Sprites made by whitewolftamer.

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
!review

File: code/modules/mob/living/carbon/human/species/outsider/undead.dm [hunk 1]
```diff
@@ -237,6 +235,17 @@
...
 	return ..()
 
+//Zombies do not have oxygen, so we have to handle the sprint this way
+/datum/species/zombie/handle_sprint_cost(var/mob/living/carbon/human/H, var/cost, var/pre_move)
+	. = ..()
+
+	if(exhaust_threshold && (H.stamina) < (exhaust_threshold * 0.8))
+		H.m_intent = M_WALK
+		H.hud_used.move_intent.update_move_icon(H)
+		to_chat(H, SPAN_DANGER("You're too exhausted to run anymore!"))
```
Comment (Generalcamo) [inline code/modules/mob/living/carbon/human/species/outsider/undead.dm:245]:
```suggestion
		to_chat(H, SPAN_WARNING("You're too exhausted to run anymore!"))
```
This is a failure state, not a damage message. Just warn the player.

File: code/__defines/species.dm [hunk 2]
```diff
@@ -39,6 +39,9 @@
 #define SPECIES_SKELETON			"Skeleton"
 #define SPECIES_CULTGHOST			"Apparition"
 #define SPECIES_ZOMBIE				"Zombie"
+#define SPECIES_ZOMBIE_BULL			"Zombie Bull"
+#define SPECIES_ZOMBIE_HUNTER		"Zombie Hunter"
+#define SPECIES_ZOMBIE_THINO		"Zombie Rhino"
```
Comment (Generalcamo) [inline code/__defines/species.dm:44]:
```suggestion
#define SPECIES_ZOMBIE_RHINO		"Zombie Rhino"
```

File: code/modules/mob/living/carbon/human/species/outsider/undead.dm [hunk 3]
```diff
@@ -313,3 +322,99 @@
...
+		melee = ARMOR_MELEE_MAJOR,
+		bomb = ARMOR_BOMB_PADDED,
+		energy = ARMOR_ENERGY_SMALL
+	)
+
+	unarmed_types = list(/datum/unarmed_attack/shocking) //This zombie cannot infect, it's an harrass type of zombie
+
+///A zombie tuned for charge attacks
+/datum/species/zombie/rhino
+	name = SPECIES_ZOMBIE_THINO
```
Comment (Generalcamo) [inline code/modules/mob/living/carbon/human/species/outsider/undead.dm:390]:
```suggestion
	name = SPECIES_ZOMBIE_RHINO
```

File: code/modules/mob/living/carbon/human/species/outsider/undead.dm [hunk 4]
```diff
@@ -313,3 +322,99 @@
...
+	name_plural = SPECIES_ZOMBIE_THINO+"s"
```
Comment (Generalcamo) [inline code/modules/mob/living/carbon/human/species/outsider/undead.dm:391]:
```suggestion
	name_plural = SPECIES_ZOMBIE_RHINO+"s"
```

File: html/changelogs/fluffyghost-zombieupgrade.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Added 3 subtypes of zombie."
+  - rscadd: "Subtype of zombie: bull, very tanky, slow moving, short sprint and low stamina."
+  - rscadd: "Subtype of zombie: hunter, very fast and sensitive to sounds, can see through walls other mobs, can leap far and often, natural climber, relatively squishy, cannot infect, attacks have a chance to electric shock, good against IPCs."
+  - rscadd: "Subtype of zombie: thino, relatively slow, can leap after a small delay and land sending people in a radius of 1 to fly except one, that gets red grabbed, punches relatively hard."
```
Comment (Generalcamo) [inline html/changelogs/fluffyghost-zombieupgrade.yml:44]:
```suggestion
  - rscadd: "Subtype of zombie: rhino, relatively slow, can leap after a small delay and land sending people in a radius of 1 to fly except one, that gets red grabbed, punches relatively hard."
```

File: html/changelogs/fluffyghost-zombieupgrade.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  - rscadd: "Zombie producing chemical grenade added."
+  - tweak: "Turning into zombie is now ~40% faster than before."
+  - tweak: "Zombies now have a limited stamina."
+  - tweak: "Normal zombies don't see through walls anymore, only the hunter do."
```
Comment (Generalcamo) [inline html/changelogs/fluffyghost-zombieupgrade.yml:48]:
```suggestion
  - balance: "Normal zombies no longer see through walls; only the hunter does."
```

**Review by Generalcamo — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species/outsider/undead.dm (hunk 1 above)
Comment (FluffyGhoster) [reply code/modules/mob/living/carbon/human/species/outsider/undead.dm:245]: This is a copied message from the function that it calls up, which uses span danger

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**